### DISCUSSION TITLE
repulsionConstant estimation

### DISCUSSION THREAD

**Discussion by mojtabaAmiraslanpour** [UvaCsl/HemoCell#41] [General discussion]
Dear Hemocell community,

I was wondering if the default value set in the config.xml file (pipeflow) for repulsionConstant is in a normal range. It is set to be 2e-22 which if in lattice units, it would correspond to a minimal force value.
```
dx = 0.5e-6
dt = 1e-7
rho = 1e3
```
Considering the above values, the default repulsion force would be on the order of 1e-30 N which is way small. I was expecting a force value in the range of 1 pN.

Am I doing this right?

**Comment by gzavo** [comment]
Hi @mojtabaAmiraslanpour,
Indeed you want something in the pN order. The best answer for now is that you will need to look at the code and follow the value of repulsionConstant to make sure how it was implemented. It has not been used actively for a while, and is mostly there to show how to introduce attractive or repulsive forces technically. We might produce a calibrated example for this in the future, but atm it is not priority in any running projects. Btw these can act between cells, or if you put down artificial repulsive particles in the wall then between cells and wall. If you happen to create an example case for this, please do consider contributing it to the repo.

**Reply by mojtabaAmiraslanpour** [reply]
Thank you @gzavo,
This is the formula I found that is implemented in the code:
![Screenshot 2022-03-27 192754](https://user-images.githubusercontent.com/15649053/160305797-67cf65b2-30bf-478b-931d-626f276e960b.png)
When `d=< r_cutOff`, this force will be activated. I have set `F_Const=0.1 pN (1.6e-5 FU)` and `r_cutOff = 1 LU`. So when particles are `0.1 LU` apart, `~1 pN` force is applied between them. It gives good results in terms of anti-penetration behavior:

![image](https://user-images.githubusercontent.com/15649053/160305928-3dda8c66-e56e-4f5b-8824-4ba1cfb6cda6.png)

However, it should not be too large which would alter the physical flow. This definitely needs more simulation cases with various repulsion forces considered to select the most efficient and correct value.

I would be more than happy to contribute to the repo if I come up with a good example and a comprehensive understanding of this!

**Comment by mojtabaAmiraslanpour** [comment]
The results of my simulations show that applying this repulsive force will prevent penetrated red blood cells to separate from each other. The definition of the force would not recognize between "penetrated" vs. "not penetrated" couple of cells and it would apply the force between any close nodes even if they are penetrated. 

Can you please elaborate on this part of the code so that I can come up with a solution? is `distance` the distance between the two nodes?

https://github.com/UvaCsl/HemoCell/blob/3189621a2118152addefd3a6f5852536c3fd2d14/core/hemoCellParticleField.cpp#L677-L743

**Comment by gzavo** [comment]
You can see how `distance` is defined in the code-segment you linked:
https://github.com/UvaCsl/HemoCell/blob/3189621a2118152addefd3a6f5852536c3fd2d14/core/hemoCellParticleField.cpp#L687
This is the distance of surface particles, with no notion of whether penetrating or not.
Checking for every particle if they are inside another cell would have a huge associated computational cost.
